src/AddImageChecks.cpp

```diff
@@ -478,8 +479,23 @@ Stmt add_image_checks(Stmt s,
             // Check the var passed in equals the constrained version (when not in inference mode)
             asserts_constrained.push_back(AssertStmt::make(var == constrained_var, error));
         }
+        if (param.defined() && param.host_alignment() > 1) {
+            string host_name = name + ".host";
+            int alignment_required = param.host_alignment();
+            Expr host_ptr = Variable::make(Int(32), host_name);
```

Comment:
Yes, I agree with you. The code in CodeGen_LLVM in the Mod visitor is unfortunate. And while an alignment that is not a power of 2 is crazy, it'll be better if our code is general enough to handle the crazies.